### PR TITLE
CORE Engine issue #521

### PR DESCRIPTION
**Pull request by dostiep**
This PR is to address CORE Engine issue #521: https://github.com/cdisc-org/cdisc-rules-engine/issues/521

### CODE REVIEW

File: tests/test_operators.py [hunk 1]
```diff
@@ -1055,33 +1055,55 @@ def test_is_not_unique_relationship(self):
...
-                "valid": ["2020-10-10", "2020-10-10", "2020-10-10", "2021", "2021", "2021", ],
-                "invalid": ["2020-10-10", None, None, "2020", "2020", None, ],
+                "USUBJID": [789, 789, 789, 789, 790, 790, 790, 790, ],
+                "SESEQ": [1, 2, 3, 4, 5, 6, 7, 8, ],
+                "SEENDTC": ["", "2006-06-10T09:47", "2006-06-17", "", "2006-06-03T10:14", "2006-06-10T10:32", "2006-06-17", "2006-06-17"],
+                "SESTDTC": ["2006-06-01", "2006-06-03T10:32", "2006-06-10T09:47", "2006-06-17", "2006-06-01", "2006-06-03T10:14", "2006-06-10T10:32", "2006-06-17"],
             }
         )
-        self.assertFalse(
-            DataframeType({"value": df}).empty_within_except_last_row({"target": "valid", "comparator": "USUBJID"})
```
Comment (gerrycampion) [inline tests/test_operators.py:1066]: The ordering property should be optional, so we should still retain the old tests that don't have the ordering property

**Review by gerrycampion — CHANGES_REQUESTED** [review]

File: business_rules/operators.py [hunk 2]
```diff
@@ -801,13 +801,14 @@ def empty(self, other_value: dict):
...
-        grouped_target = self.value.groupby(comparator)[target]
+        ordered_df = self.value.sort_values(by=[comparator, order_by_column])
+        grouped_target = ordered_df.groupby(comparator)[target]
         # validate all targets except the last one
         results = grouped_target.apply(lambda x: x[:-1]).apply(lambda x: x in ["", None])
-        # extract values with corresponding indexes from results
-        self.value[f"result_{uuid4()}"] = results.reset_index(level=0, drop=True)
-        return True in results.values
+        # return values with corresponding indexes from results
+        return pd.Series(results.reset_index(level=0, drop=True))
```
Comment (gerrycampion) [inline business_rules/operators.py:811]: We should add documentation for the added param https://github.com/cdisc-org/cdisc-rules-engine/blob/main/resources/schema/Operator.md#empty_within_except_last_row

File: tests/test_operators.py (hunk 1 above)
Comment (dostiep) [reply tests/test_operators.py:1066]: OK done

File: business_rules/operators.py (hunk 2 above)
Comment (dostiep) [reply business_rules/operators.py:814]: Done, see PR on rules-engine.

File: business_rules/operators.py
[the remark this answers is not in the document]
Comment (dostiep) [reply business_rules/operators.py:1]: Done

**Comment by dostiep** [comment]
All comments were addressed and a PR was raised also on rules-engine

**Comment by gerrycampion** [comment]
Code changes copied over and replaced by: https://github.com/cdisc-org/cdisc-rules-engine/pull/703

**Pull request closed without merging**